### PR TITLE
Refactor and improve make_fx testing

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #103198
* #103197
* __->__ #103196

This is in preparation for the custom_op_compile_check utility, which
will call the newly refactored function.

This PR:
- splits off code into helper functions
- adds clearer error messages
- stops updating the inputs destructively (leading to slightly slower
tests)

### CODE REVIEW

File: torch/testing/_internal/optests/make_fx.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    def run(f, *args, **kwargs):
+        return wrapper_set_seed(f, *args, **kwargs)
+
+    traced_f = make_fx(f, tracing_mode=tracing_mode)(*new_args)
+
+    msg = (
+        "op(*args, **kwargs) and make_fx(op)(*args, **kwargs) produced different "
+        "values. This could mean that your abstract impls (meta/FakeTensor impls) "
+        "are incorrect, that your operator is not completely traceable (e.g., "
+        "it relies on some global state), or that there is a bug in make_fx."
```
Comment (bdhirsh) [inline torch/testing/_internal/optests/make_fx.py:26]: is it worth including the traced graph in the error message for sanity?

Comment (zou3519) [reply torch/testing/_internal/optests/make_fx.py:26]: we could!

File: test/test_proxy_tensor.py [hunk 2]
```diff
@@ -1566,8 +1567,6 @@ def f(a, b, c, d, e):
     xfail('float_power', ''),  # base given to float_power_ has dtype Float but the operation's result requires dtype Double
     # decomp not implemented
     xfail('unique', ''),
-    # in-place has a different signature than out-of-place
-    xfail('uniform', ''),
```
[the remark this answers is not in the document]
Comment (zou3519) [reply test/test_proxy_tensor.py:1570]: Previously, we were setting the random seed for trace_f(*args) but not setting it for f(*args). This PR changes it so that we set the random seed for both. Because random operators are run in the same order, we can expect the output of both of these to be the same.

File: torch/testing/_internal/optests/make_fx.py (hunk 1 above)
Comment (zou3519) [reply torch/testing/_internal/optests/make_fx.py:26]: On second thought, I don't want to, because it means the error message might get long, especially if you have a CompositeImplicitAutograd op that decomposes into something like matmul. It's pretty easy for the user to do the make_fx(op)(*args, kwargs) themselves and see what it traces out as.

File: test/test_proxy_tensor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bdhirsh) [reply test/test_proxy_tensor.py:1570]: oof, nice catch

**Review by bdhirsh — APPROVED** [review]

File: torch/testing/_internal/optests/make_fx.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    return f, args, kwargs, extra_args, extra_kwargs
+
+
+def randomize(args):
+    def transform(x):
+        if not x.dtype.is_floating_point:
+            return x
+        with torch.no_grad():
+            y = x.detach().clone().uniform_(0, 1)
```
Comment (soulitzer) [inline torch/testing/_internal/optests/make_fx.py:83]: nit: you can do this in one line :)
`return x.detach().clone().uniform_(0, 1).requires_grad_(x.requires_grad)`

(no need to have both no-grad and detach)

**Review by soulitzer — APPROVED** [review]

File: torch/testing/_internal/optests/make_fx.py (hunk 3 above)
Comment (zou3519) [reply torch/testing/_internal/optests/make_fx.py:83]: Good point!
